Factory/PhpredisClientFactory.php

```diff
@@ -44,6 +44,9 @@ public function create($class, array $dsns, $options, $alias)
             throw new \LogicException(sprintf('The factory can only instantiate \Redis|\RedisCluster classes: "%s" asked', $class));
         }
 
+        // Normalize the DSNs, because using processed environment variables could lead to nested values.
+        $dsns = count($dsns) === 1 && is_array($dsns[0]) ? $dsns[0] : $dsns;
```

Comment:
Is this the way it is managed in other bundle or that's more like a hack? 

Should we fix the config of the bundle instead?

Reply:
This is more like a hack. Unfortunately I didn't find any other way to solve this…

When we write this in the config :

```
dsn: "%env(csv:REDIS_URL)%"
```

The `SncRedisExtension` class only sees an array with a single string, representing an env var placeholder. There's only the `PhpredisClientFactory` class which can see the final values.

Reply:
ok thank you!

Reply:
Oooh, maybe we could solve this directly in the config with [`ContainerBuilder::resolveEnvPlaceholders()`](https://github.com/symfony/symfony/blob/0dce5cf0ba439450475519137541e7f48894852f/src/Symfony/Component/DependencyInjection/ContainerBuilder.php#L1352)…

I will try to look for a solution with this method.

Reply:
No because it would mean the cache is built with resolved environment instead of resolving them at runtime. We already bugs related to this behavior.

Reply:
Nevermind then. Lets keep the hacky thing. 🙂